app/controllers/teams_controller.rb

```diff
@@ -6,11 +6,19 @@ class TeamsController < ApplicationController
   load_and_authorize_resource except: [:index, :show]
 
   def index
```

Comment:
Oh, no worries 💚 

First of all I'd try to get rid of redundant queries. So a super simple thing here could for instance be to move the conditional:
```ruby
def index
  # stuff
  if year.present?
    @teams = query1
  else
    @teams = query2
  end
end
```

Then move the two queries from the second assignment into one:
```ruby
@teams = Team
  .by_season(year)
  .accepted
  .includes(:activities)
  .order("teams.kind, activities.created_at #{direction}")
  .references(:activities)
```

Can you use this as a starting point?